### ISSUE TITLE
Add a sitemap submission feature

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#7048]
## Feature Description

Now that WordPress has sitemap functionality built in, make it easy for users to submit this to Search Console, or for them to submit their own other than the default WordPress sitemap. This could include sitemaps built via other plugins. 

An example of what may work would be the below, however, this would need some consideration during Site Kit setup, so users are aware or have the choice of submitting their sitemap (rather than having to visit their Search Console settings)

![sitemap](https://github.com/google/site-kit-wp/assets/41326532/cd80b6fe-8b18-41be-81f5-e41799f21386)

Note that there were previous feature requests for the same, see below:
- https://github.com/google/site-kit-wp/issues/1145
- https://github.com/google/site-kit-wp/issues/1535

In the first previous feature request above it was closed on the based that WordPress already has default sitemap functionality built in. This however doesn't mean Search Console recognizes the sitemap. Sitemap submission is also [supported by the API in use](https://developers.google.com/webmaster-tools/v1/sitemaps).

We've also had numerous support requests or queries for this, see some below examples:
- https://wordpress.org/support/topic/google-site-kit-sitemap-and-search-console/
- https://wordpress.org/support/topic/showing-page-not-found-inside-sitemap_index-xml/
- https://wordpress.org/support/topic/google-search-console-sitemap-2/

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by marrrmarrr** [comment]
Hmmm I think this is going into replicating too much of Search Console functionality territory. Also I think a lot of SK users probably don't know what Sitemaps are, so it would probably end up as little used element hidden in Settings. 
I think Sitemaps could be an interesting tool for a future feature where we help people immediately submit their new posts, or submit updated posts.

**Comment by aaemnnosttv** [comment]
> Hmmm I think this is going into replicating too much of Search Console functionality territory. Also I think a lot of SK users probably don't know what Sitemaps are, so it would probably end up as little used element hidden in Settings.

I would agree if this required the kind of input envisioned above, however that wouldn't really be necessary since WP core's sitemap URL is available to us as of WP 5.5.1, so we could technically auto-submit this to Search Console during setup – no added interface or action required.

It's worth noting that WP also already makes its sitemaps discoverable automatically via its `robots.txt` so it's not clear how much, if any, additional benefit this submission would provide. We've had additional requests for this feature lately, but upon further investigation, it may not be needed afterall.

<img width="1025" alt="image" src="https://github.com/user-attachments/assets/76e0f61f-b3c8-4f36-a3fd-bab91d29aaed">

Source: https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap#cmssitemap

Reopening for further consideration while we investigate.

**Comment by ajtruckle** [comment]
It would be nice to make it easier to just use the WP sitemap.

**Comment by benbowler** [comment]
I would use this too for my personal blog.

**Comment by aaemnnosttv** [comment]
> It would be nice to make it easier to just use the WP sitemap.

@ajtruckle that is essentially how it works now automatically as noted above 🙂 
WordPress core adds a reference to its sitemap in the `robots.txt` file, which is one of the supported methods for making all search engines aware of the sitemap, not just Google. While this is not the same as "submitting" the sitemap in Search Console, search engines will be aware of it.

The main question we are looking to answer to decide whether this is worth doing is based on whether or not there is any additional benefit in submitting the sitemap, rather than discovering it at crawl time as it is now. Note: there is a difference between submitting a sitemap (the URL of the resource) and crawling it. Just because it's submitted, doesn't mean it will be crawled any faster.

**Comment by aaemnnosttv** [comment]
Closing after additional investigation, as there doesn't appear to be a substantial benefit for submitting a sitemap via Search Console. The documentation clearly lists [`robots.txt` as a valid form of sitemap submission](https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap#cmssitemap:~:text=Insert%20the%20following%20line%20anywhere%20in%20your%20robots.txt%20file%2C%20specifying%20the%20path%20to%20your%20sitemap.%20We%20will%20find%20it%20the%20next%20time%20we%20crawl%20your%20robots.txt%20file) which all sites have since WP 5.5 (which will be 6 years old in a few days).

Submitting a sitemap in Search Console is only really necessary for us if we wanted to surface information about the sitemap or its contents in SK which is always something we could reconsider in the future.
